devito/ir/stree/algorithms.py

```diff
@@ -1,75 +1,61 @@
+from collections import defaultdict
+from itertools import groupby
+
```

Comment:
why empty line?

Reply:
it's the standard adopted throughout the entirety of devito

<imports from stdlib>
<blank line>
<imports from third-party deps>
<blank line>
<local imports>
...

Reply:
probably, they might have escaped reviewing, fixable as we encounter them